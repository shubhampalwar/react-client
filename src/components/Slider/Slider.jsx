import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { DEFAULT_BANNER_IMAGE } from '../../configs/constants';
import { getRandomNumber, getNextRoundRobin } from '../../lib/utils/math';

class Slider extends Component {
  constructor(props) {
    super(props);
    this.state = {
      index: 0,
    };
  }

  componentWillMount() {
    clearInterval(this.interval);
  }

  componentDidMount() {
    const { random, duration, banners } = this.props;
    this.interval = setInterval(() => {
      const { index } = this.state;
      return (random) ? this.setState({ index: getRandomNumber(banners.length) })
        : this.setState({ index: getNextRoundRobin(banners.length, index) });
    }, duration);
  }

  render() {
    const { index } = this.state;
    const {
      altText, banners, defaultBanner, ...rest
    } = this.props;
    const source = (banners.length === 0) ? defaultBanner : banners[index];
    return (
      <>
        <div style={{ textAlign: 'center' }}>
          <img src={source} alt={altText} style={{ height: rest.height }} />
        </div>
      </>
    );
  }
}
Slider.propTypes = {
  altText: PropTypes.string,
  banners: PropTypes.arrayOf(PropTypes.string),
  defaultBanner: PropTypes.string,
  duration: PropTypes.number,
  height: PropTypes.string,
  random: PropTypes.bool,
};
Slider.defaultProps = {
  altText: 'Default Banner',
  banners: [],
  defaultBanner: DEFAULT_BANNER_IMAGE,
  duration: 2000,
  height: '200px',
  random: false,
};

export default Slider;
