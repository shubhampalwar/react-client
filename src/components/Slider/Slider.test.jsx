import { shallow } from 'enzyme';
import React from 'react';
import Slider from './Slider';

const wrapper = shallow(<Slider />);

describe('Slider', () => {
  it('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
