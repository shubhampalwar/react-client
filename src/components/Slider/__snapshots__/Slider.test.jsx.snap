// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Slider
    altText="Default Banner"
    banners={Array []}
    defaultBanner="/default.png"
    duration={2000}
    height="200px"
    random={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <img
          alt="Default Banner"
          src="/default.png"
          style={
            Object {
              "height": "200px",
            }
          }
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="Default Banner"
          src="/default.png"
          style={
            Object {
              "height": "200px",
            }
          }
        />,
        "style": Object {
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Default Banner",
          "src": "/default.png",
          "style": Object {
            "height": "200px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <img
            alt="Default Banner"
            src="/default.png"
            style={
              Object {
                "height": "200px",
              }
            }
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Default Banner"
            src="/default.png"
            style={
              Object {
                "height": "200px",
              }
            }
          />,
          "style": Object {
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Default Banner",
            "src": "/default.png",
            "style": Object {
              "height": "200px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
